src/modules/dashboard/components/CreatePaymentDialog/CreatePaymentDialogForm.tsx

```diff
@@ -0,0 +1,394 @@
+
+  const fromDomainTokenBalance = useMemo(() => {
+    const token =
+      tokenBalancesData &&
+      tokenBalancesData.tokens.find(({ address }) => address === tokenAddress);
+    const from = getBalanceFromToken(token, fromDomain);
+    return from;
+  }, [fromDomain, tokenAddress, tokenBalancesData]);
+
+  useEffect(() => {
+    const errors: {
+      amount?: any;
+    } = {};
+
+    if (!selectedToken || !(amount && amount.length)) {
+      errors.amount = undefined; // silent error
+    } else {
+      const convertedAmount = bigNumberify(
+        moveDecimal(
+          amount,
+          getTokenDecimalsWithFallback(selectedToken.decimals),
+        ),
+      );
+      if (!convertedAmount.eq(0)) {
+        errors.amount = MSG.noAmount;
+      } else if (
+        fromDomainTokenBalance &&
+        fromDomainTokenBalance.lt(convertedAmount)
+      ) {
+        errors.amount = MSG.noBalance;
+      }
+    }
+
+    setErrors(errors);
+  }, [
+    amount,
+    fromDomain,
+    fromDomainRoles,
+    fromDomainTokenBalance,
+    selectedToken,
+    setErrors,
+  ]);
+
+  const userHasFundingPermission = userHasRole(
+    fromDomainRoles,
+    ColonyRole.Funding,
+  );
+  const userHasAdministrationPermission = userHasRole(
+    fromDomainRoles,
+    ColonyRole.Administration,
+  );
+  const userHasPermission =
+    userHasFundingPermission && userHasAdministrationPermission;
+  const requiredRoles: ColonyRole[] = [
+    ColonyRole.Funding,
+    ColonyRole.Administration,
+  ];
+
+  return (
+    <>
+      <FormStatus status={status} />
+      <DialogSection>
+        <Heading
+          appearance={{ size: 'medium', margin: 'none' }}
+          text={MSG.title}
+          className={styles.title}
+        />
+      </DialogSection>
+      {!userHasPermission && (
+        <DialogSection>
+          <PermissionRequiredInfo requiredRoles={requiredRoles} />
+        </DialogSection>
+      )}
+      <DialogSection>
+        <div className={styles.domainSelects}>
+          <div>
+            <Select
+              options={domainOptions}
+              label={MSG.from}
+              name="fromDomain"
+              appearance={{ theme: 'grey', width: 'fluid' }}
+              disabled={!userHasPermission}
+            />
+            {!!tokenAddress && (
+              <div className={styles.domainPotBalance}>
+                <FormattedMessage
+                  {...MSG.domainTokenAmount}
+                  values={{
+                    amount: (
+                      <Numeral
+                        appearance={{
+                          size: 'small',
+                          theme: 'grey',
+                        }}
+                        value={fromDomainTokenBalance || 0}
+                        unit={getTokenDecimalsWithFallback(
+                          selectedToken && selectedToken.decimals,
+                        )}
+                        truncate={3}
```

Comment:
I'm not sure why we truncate in here. I have a feeling It's not really correct.

Anyway, just leave it like it is for now.